kernel/src/process.rs

```diff
@@ -84,6 +69,130 @@ pub unsafe fn load_processes(
     }
 }
 
+/// This trait is implemented by process structs.
+pub trait ProcessType {
+    /// Queue a callback for the process. This will be added to a per-process
+    /// buffer and passed to the process by the scheduler.
+    fn schedule(&self, callback: FunctionCall) -> bool;
+
+    /// Queue an IPC operation for this process.
+    fn schedule_ipc(&self, from: AppId, cb_type: IPCType);
```

Comment:
Why can `schedule` fail (assume the bool is fail), but `schedule_ipc` not?

Reply:
Because it's a bug in the original code I wrote. `schedule_ipc` should also be returning an error/bool, and the consumer (in `ipc.rs` should be returning a corresponding error to the calling process.